dask/array/chunk.py

```diff
@@ -265,6 +265,12 @@ def arange(start, stop, step, length, dtype, like=None):
     return res[:-1] if len(res) > length else res
 
 
+def linspace(start, stop, num, endpoint=True, dtype=None):
+    return np.linspace(
+        float(start), float(stop), int(num), endpoint=endpoint, dtype=dtype
```

Comment:
We don't currently have tests for this, but this breaks the usage of `np.linspace` with `__array_function__` dispatching because we're now casting to `float`, rather than ndarray types. I think the best would be to `.compute()` the input arrays here if they are Dask arrays, in that case we would still respect the `__array_function__` dispatching by using the underlying ndarray instead of `float`, in that case we would still be able to generate chunks respecting the `meta` type of the Dask array.

Any objections to resolving this the way I proposed above @jrbourbeau ?

Reply:
Thanks for the feedback. How should I check if the input arrays are Dask arrays? I cannot import Array
from **chunk.py** because it creates a circular dependency with core, therefore I cannot use `isinstance(array, Array)`. I see two alternatives:

1. Leave the function in **chunk.py** and use has [hasattr](https://docs.python.org/3/library/functions.html#hasattr) to check if it has a `compute` attribute (another possibility is to wrap the function call in a try-catch block)
2. Move this function to **creation.py**, name it `_linspace_chunk` (tentative name) and call `isinstance(array, Array)`

I prefer the second option. Do you see any other alternatives? @pentschev @jrbourbeau

Reply:
You can import `Array` within the function itself to avoid the circular dependency. We do that in a few places, such as [here](https://github.com/dask/dask/blob/main/dask/array/slicing.py#L875), then checking for `isinstance(array, Array)` would probably suffice.

Reply:
@pentschev Thanks for the tip! I did as suggested, so I'm resolving the conversation

Reply:
Thanks @mesejo , I've verified that this also works with `__array_function__` locally.